### ISSUE TITLE
slice parameter shadows built-in slice function in container.py

### ISSUE THREAD

**Issue by Iago-7** [decryptofy/scarr#21]
In the container.py file located within the container directory, the list parameter named “slice” in the __init__ method shadows the name of the built-in “slice” function in Python. If you were to use the slice function in the same method in a future update of the method, an error would be raised. I think renaming the slice parameter for better clarity, and to prevent a potential future error might be beneficial.

I'd be more than happy to follow up with a pull request to fix it.

**Comment by decryptofy** [comment]
Good catch. We should probably rename 'slice' to 'slab' (inspired by the HDF5 terminology of 'hyperslab'). Please note that we need to refactor all occurrences in the code plus the linked Jupyter notebooks.

- please prepare a PR for the needed refactoring
- this PR should add your name to the list of contributors (see README.md)

**Comment by Iago-7** [comment]
Sounds great! I'll start working on that. Would you prefer time_slice to become time_slab as well, or remain time_slice?

**Comment by decryptofy** [comment]
Hmm, I see that our naming in that area can be improved. First priority is to keep it consistent (i.e., changing 'slice' in the init method to 'slab', should result in replacing the word slice consistently throughout, i.e., changing time_slice into time_slab).

Quickly looking further into the code, it is apparent that we should further improve clarity of the naming, as there is another variable called slice_index that we have to take care of, too. I think this is how we should proceed:

- time_slice --> slab_range
- slice_index --> slab_points

slab_range[0] and slab_range[1] will be the start:stop index, whereas slab_points should be a list of points.

After this change, slice() only remains within the scope of the built-in function. The other variables we can still refactor if needed, as their use is quite limited in the code base.

**Comment by Iago-7** [comment]
It's also found in two of the Jupyter notebook files, but just as slice, so that should be an easy fix.

**Comment by decryptofy** [comment]
> Slice is also found in trace_handler.py which has self.sample_slice, self.slices, slices, and batch_slice, and correlation_data.py which has self.slices. How would you prefer naming with those?

in all these cases, I would substitute slice(s) with slab(s).

We/I should probably refactor some other things (batchSize -> batch_size, Async -> async) to use proper PEP8 naming conventions while we are at it, and then increase the version number to 0.1.1 as all these changes break compatibility with previous versions. Time to start a changelog.

**Comment by Iago-7** [comment]
Alright I'm on it!

Some other PEP8 issues I noticed are lines that are too long and having an extra space on either side of a parameter and default argument.

Also, is it an issue having mutable default arguments? From what I remember, if you call the methods with them more than once it'll only use the original object created from the first call. I'm a student at OSU, and still learning, so any insight there would be helpful.

**Comment by Iago-7** [comment]
I appreciate the feedback -- the desire for best performance possible makes a lot of sense. If there was an issue with mutables, I'm sure you would have probably come across it by now.

**Comment by decryptofy** [comment]
All done; good job. Closed.
